### ISSUE TITLE
Use Azure Blob Storage for image upload

### ISSUE THREAD

**Issue by linkdotnet** [linkdotnet/Blog#364]
I am working on the image upload function. Technically, this is very straightforward, and I will start with Azure Blob Storage.

There are some open questions, which might require some more polishing:
1. Do we take the filename of the uploaded file one-to-one?
2. If so - that could lead to conflicts. Therefore: Should we just add some small random hash at the end of each file? Does that have implications on SEO?
3. Do we just use a random guide and basically never run into a collision?

My preferred option is right now: (2)
So basically, if a user uploads a file via Drag 'n Drop, we take that filename and append a random 6 to 8 digit hash at the end to avoid any conflicts.

@ncosentino Do you have any preference here?

**Comment by ncosentino** [comment]
My use case, when I migrate, will be that I want to enforce 1:1 naming if
possible in my URLs. I'm not actually convinced I need to keep image URLs
the same to be honest, but it's a fear of mine that I might have something
out there expecting an image at some path and it'll be broken.

(Might be just a false fear I have)

I'm fine with two, generally. Any opportunity for the new guids with some
time element in them?

**Comment by linkdotnet** [comment]
I do think I have a version that will gives all the flexibility you want:
![Feature](https://github.com/user-attachments/assets/2a839eb4-cb9f-4f1f-b570-5d80e9ce4abe)

Basically it allows you to define the whole path. While kind of an MVP, it circumnavigate a whole lot of edge cases. 

While "labour" - it covers your case.

**Comment by ncosentino** [comment]
Looks awesome!

**Comment by linkdotnet** [comment]
@ncosentino 
I merged the feature. It is basically the version above with some extras:
1. Cache-control header can be automatically set to a week (for SEO).
2. Optionally, an Azure CDN can be configured!

The basic setup looks like this: Adapt the following in your `appsettings.json`:
```json
{
  ...
  "ImageStorageProvider": "Azure",
  "ImageStorage": {
    "AuthenticationMode": "Default", // Can Also be "ConnectionString"
    "ConnectionString": "", // If AuthMode is ConnectionString, here should be the ConnectionString from Azure
    "ServiceUrl": "",  // The Service Url - only used when AuthMode is Default
    "ContainerName": "" // Name of the container
  }
}
```

And there you go. The whole documentation can be found here: [Media Upload](https://github.com/linkdotnet/Blog/tree/master/docs/Media/Readme.md)
